src/preset.ts

```diff
@@ -445,17 +600,53 @@ export async function expandConfigurePreset(folder: string,
 
   expansionOpts.envOverride = expandedPreset.environment as EnvironmentVariables;
 
+  const presetsFile = getPresetsFile(folder);
+  if (presetsFile && presetsFile.version >= 3) {
+    // For presets v3+ binaryDir and generator are optional, but cmake-tools needs a value. Default to something reasonable.
+    if (!preset.binaryDir) {
+      const defaultValue =  '${sourceDir}/out/build/${presetName}';
+
+      log.warning(localize('binaryDir.undefined', 'Configure preset {0}: No binaryDir specified, using default value "{1}"', preset.name, defaultValue));
```

Comment:
We get complaints from customers when we warn too frequently.  If this is going to be printed every time the presets file is parsed, it might be too much. For cases like this our options are to track whether we've warned this or not already and only do it once, or to demote the warning to `log.debug`.

Reply:
That's fair, I can make it a debug warning.